drake_ros_introspection/include/drake_ros_introspection/predicates/declared_by.h

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <drake/systems/framework/system_base.h>
+
+namespace drake_ros_introspection {
+namespace predicates {
+
+namespace internal {
+
+template <typename BaseT, typename T>
+bool isinstance(const T& value) {
```

Comment:
`constexpr bool` since this is a compime check?

Reply:
Not a compile time check. This function is used to detect whether a `SystemBase` type instance is actually an (upcast) instance of some specific `LeafSystem`, and that can only be done in runtime.